### ISSUE TITLE
DeepPartial<T> in value, valueChanges and patchValue()

### ISSUE THREAD

**Issue by kbrilla** [no0x9d/ngx-strongly-typed-forms#15]
The signature in value and valueChanges is using type T, but should't it use DeepPartial<T>. Same with for example patchValue. 
value and valueChagnes extract fields that are disabled so it's returning deep partial type. And patchValue takes incomplete object tree so deepPartial again. 

What's your point of view on that? If You agree I could add that change.

> When Angular uses TypeScript 2.8 the return value should be changed.

We already have it since like forever now :)

**Comment by no0x9d** [comment]
Yes TypeScript 2.8 ist long there, but I have to admit that I didn't think this feature was too important for the little bit of spare time I have.
Now I realised, that I can't event use my lib on the default reactive forms example from angular.io documentation, because it uses a FormGroup#patch with a deep partial value.

If you still want to add it, I'm happy to accept a pull request.
